cmake_ext.py

```diff
@@ -128,22 +129,26 @@ def build_extension(self, ext):
         if cmake_cache.exists():
             with cmake_cache.open() as f:
                 s = f.read()
-                cached_generator = cache_value("CMAKE_GENERATOR", s)
-                cached_cfg = cache_value("CMAKE_BUILD_TYPE", s)
-                disagreement = []
-                for arg in cmake_args:
-                    if arg.startswith("-D"):
-                        key, value = arg[2:].split("=")
-                        cached_value = cache_value(key, s)
-                        disagreement.append(value != cached_value)
-                if any(
-                    disagreement
-                    + [
-                        (cmake_generator and cached_generator != cmake_generator),
-                        cached_cfg != cfg,
-                    ]
-                ):
-                    cmake_cache.unlink()
+            cached_generator = cache_value("CMAKE_GENERATOR", s)
+            cached_cfg = cache_value("CMAKE_BUILD_TYPE", s)
+            disagreement = []
+            for arg in cmake_args:
+                if arg.startswith("-D"):
+                    key, value = arg[2:].split("=")
+                    cached_value = cache_value(key, s)
+                    # Change \ to / in case of Windows
+                    disagreement.append(
+                        value.replace("\\", "/") != cached_value.replace("\\", "/")
+                    )
```

Comment:
Please use pathlib.Path's to abstract this away.

Reply:
Ok. I found this solution that does the same thing:

```python
PureWindowsPath(value).as_posix() != PureWindowsPath(cached_value).as_posix()
```

Is it fine?